runtime/sema/type.go

```diff
@@ -6436,3 +6469,227 @@ func (t *CapabilityType) GetMembers() map[string]MemberResolver {
 		},
 	})
 }
+
+type BuiltinCompositeType struct {
```

Comment:
I had basically at the same time also introduced the same as `NominalType`. Could you please switch over to using that one instead and remove this new one? Sorry, I should have coordinated that better

Reply:
That's a really good point. I also had the same thought at one point. But the catch was, there are a couple of places that we switch on the type `BuiltinCompositeType` to go in different paths. So it's sort of useful to have a separate type for built-in composites. (However, this `BuiltinCompositeType` might well be temporary, because of the below reason.)

Also, I realized that, as we migrate the rest of the types (e.g: auth account, public account, etc), and maybe for making future proof (leaving space to add builtin interfaces, contracts, etc. in the future), we could actually use the existing `CompositeType` and `CompositeValue` to represent BuiltinComposite types as well, eventually. So the checker and runtime only know one composite type/value - it doesn't matter where it comes from. 

Then we could also remove some of the fields (like `Members`, `nestedTypes`, etc) from the `NominalType` and use it purely for truly-nominal types (like `Int`, `String`, etc.).  
WDYT?